core/services/fluxmonitorv2/integrations_test.go

```diff
@@ -0,0 +1,946 @@
+	awaitSubmission(t, submissionReceived)
+
+	// change in price should trigger run
+	reportPrice = int64(4)
+	awaitSubmission(t, submissionReceived)
+
+	// raise both flags
+	fa.flagsContract.RaiseFlag(fa.sergey, fa.aggregatorContractAddress)
+	fa.flagsContract.RaiseFlag(fa.sergey, utils.ZeroAddress)
+	fa.backend.Commit()
+
+	// wait for FM to receive flags raised logs
+	assert.Eventually(t, func() bool {
+		ilogs, err := fa.flagsContract.FilterFlagRaised(nil, []common.Address{})
+		require.NoError(t, err)
+		logs := cltest.GetLogs(t, nil, ilogs)
+		return len(logs) == 4
+	}, 7*time.Second, 100*time.Millisecond)
+
+	// change in price should not trigger run
+	reportPrice = int64(8)
+	assertNoSubmission(t, submissionReceived, 5*time.Second, "should not trigger a new run, while flag is raised")
+}
+
+func TestFluxMonitor_InvalidSubmission(t *testing.T) {
+	// 8 decimals places used for prices.
+	fa := setupFluxAggregatorUniverse(t, WithMinMaxSubmission(
+		big.NewInt(100000000),     // 1 * 10^8
+		big.NewInt(1000000000000), // 10000 * 10^8
+	))
+
+	oracleList := []common.Address{fa.neil.From, fa.ned.From, fa.nallory.From}
+	_, err := fa.aggregatorContract.ChangeOracles(fa.sergey, emptyList, oracleList, oracleList, 1, 3, 2)
+	assert.NoError(t, err, "failed to add oracles to aggregator")
+	fa.backend.Commit()
+
+	// Set up chainlink app
+	app := setupApplication(t, fa, func(cfg *orm.Config) {
+		cfg.Set("DEFAULT_HTTP_TIMEOUT", "100ms")
+		cfg.Set("TRIGGER_FALLBACK_DB_POLL_INTERVAL", "1s")
+		cfg.Set("MIN_OUTGOING_CONFIRMATIONS", "2")
+		cfg.Set("MIN_OUTGOING_CONFIRMATIONS", "2")
+		cfg.Set("ETH_HEAD_TRACKER_MAX_BUFFER_SIZE", "100")
+	})
+	require.NoError(t, app.StartAndConnect())
+
+	// Report a price that is above the maximum allowed value,
+	// causing it to revert.
+	reportPrice := int64(10001) // 10001 ETH/USD price is outside the range.
+	mockServer := cltest.NewHTTPMockServerWithAlterableResponse(t,
+		generatePriceResponseFn(&reportPrice),
+	)
+	t.Cleanup(mockServer.Close)
+
+	// Generate custom TOML for this test due to precision change
+	toml := `
+type              = "fluxmonitor"
+schemaVersion     = 1
+name              = "example flux monitor spec"
+contractAddress   = "%s"
+precision = %d
+threshold = 0.5
+absoluteThreshold = 0.01
+
+idleTimerPeriod = "1h"
+idleTimerDisabled = false
+
+pollTimerPeriod = "%s"
+pollTimerDisabled = false
+
+observationSource = """
+ds1 [type=http method=GET url="%s"];
+ds1_parse [type=jsonparse path="data,result"];
+
+ds1 -> ds1_parse
+"""
+`
+
+	s := fmt.Sprintf(toml, fa.aggregatorContractAddress, 8, "100ms", mockServer.URL)
+
+	// raise flags
+	fa.flagsContract.RaiseFlag(fa.sergey, utils.ZeroAddress) // global kill switch
+	fa.flagsContract.RaiseFlag(fa.sergey, fa.aggregatorContractAddress)
+	fa.backend.Commit()
+
+	requestBody, err := json.Marshal(models.CreateJobSpecRequest{
+		TOML: string(s),
+	})
+	assert.NoError(t, err)
+
+	j := cltest.CreateJobViaWeb2(t, app, string(requestBody))
+
+	go func() {
```

Comment:
I think we want to clean up this goroutine when the test exists no? i.e. 
```
	tick := time.NewTicker(1 * time.Second)
	defer tick.Stop()
	go func() {
		for range tick.C {
			b.Commit()
		}
	}()

```

Reply:
Fixed